shark_turbine/kernel/wave/codegen.py

```diff
@@ -744,16 +744,20 @@ def handle_reduction(emitter: WaveEmitter, node: fx.Node):
     start = arith_d.constant(IndexType.get(), int(0))
 
     idxc = IndexingContext.current()
+    tile_size = None
+    for constraint in emitter.constraints:
+        if isinstance(constraint, TilingConstraint) and constraint.dim == axis:
+            tile_size = constraint.tile_size.subs(idxc.subs)
+    assert tile_size is not None, "Could not find tiling constraint for reduction axis."
+
     # For now, we assume that dimensions that have tiling constraints on them,
     # do not have any other constraints.
     dim = axis.subs(idxc.subs)
-    end = arith_d.constant(IndexType.get(), int(dim))
+    end = arith_d.constant(IndexType.get(), int(dim // tile_size))
 
-    step = None
-    for constraint in emitter.constraints:
-        if isinstance(constraint, TilingConstraint) and constraint.dim == axis:
-            tile_size = constraint.tile_size.subs(idxc.subs)
-            step = arith_d.constant(IndexType.get(), int(tile_size))
+    # Since we divide the end by the tile size, we need to make sure that the
+    # step is 1.
+    step = arith_d.constant(IndexType.get(), int(1))
```

Comment:
Nice fix/refactor! seems like we can do this because we already take tile_size into account for index computation during expansion phase?

Reply:
yes, we compute the index as induction_var * tile_size so by doing this, we don't double multiply.